redis_reservation/__init__.py

```diff
@@ -23,6 +24,16 @@ def __init__(self, redis, key, by, lock_ttl=30*60, heartbeat_interval=10*60):
     self.reserved = False
     self.logger = logging.getLogger(by)  # log with the name of the reserver
     self.heartbeat_thread = None
+    self.previous_sigterm_handler = signal.getsignal(signal.SIGTERM)
+    signal.signal(signal.SIGTERM, self.sigterm_handler)
+
+  def sigterm_handler(self, signum, frame):
+    try:
+      self.logger.info("RESERVE: caught sigterm, releasing reservation")
+      self.release()
+    finally:
```

Comment:
Just making sure I understand what's going on here - this is for the case that there was an additional signal while processing the current signal? Can `self.release` cause `SIGTERM`?

Reply:
`signal.signal(...)` will override any previously set signal handler: https://docs.python.org/2/library/signal.html#signal.signal

this `finally` makes sure to call any previously-set signal handler after releasing the reservation